src/components/node.js

```diff
@@ -307,24 +349,43 @@ class Node extends React.Component<INodeProps, INodeState> {
     const nodeSubtypeXlinkHref =
       Node.getNodeSubtypeXlinkHref(data, nodeSubtypes) || '';
 
-    // get width and height defined on def element
-    const defSvgNodeElement: any = nodeTypeXlinkHref
-      ? document.querySelector(`defs>${nodeTypeXlinkHref}`)
-      : null;
-    const nodeWidthAttr = defSvgNodeElement
-      ? defSvgNodeElement.getAttribute('width')
-      : 0;
-    const nodeHeightAttr = defSvgNodeElement
-      ? defSvgNodeElement.getAttribute('height')
-      : 0;
-
-    props.width = nodeWidthAttr ? parseInt(nodeWidthAttr, 10) : props.width;
-    props.height = nodeHeightAttr ? parseInt(nodeHeightAttr, 10) : props.height;
+    props.xlinkHref = nodeTypeXlinkHref;
+    props.className = nodeClassName;
+    props['data-index'] = index;
+    props.x = -props.width / 2;
+    props.y = -props.height / 2;
 
     if (renderNode) {
+      // Add this classname to the rendered node props as it is used by the
+      // `edge.js` intersection detection logic
+      props.className = `${props.className} digraph-foreign-node-${data[nodeKey]}`;
```

Comment:
Should `digraph-foreign-node-${data[nodeKey]}` only be added if foreign node is used?